apps/client/src/services/index.ts

```diff
@@ -26,42 +54,30 @@ export function setServiceUrl(url: string) {
 }
 
 // Dashboard API
-export type ListDashboards = typeof DashboardsService.dashboardsControllerList;
-export type CreateDashboard =
-  typeof DashboardsService.dashboardsControllerCreate;
-export type ReadDashboard = typeof DashboardsService.dashboardsControllerRead;
-export type UpdateDashboard =
-  typeof DashboardsService.dashboardsControllerUpdate;
-export type DeleteDashboard =
-  typeof DashboardsService.dashboardsControllerDelete;
-export type BulkDeleteDashboards =
-  typeof DashboardsService.dashboardsControllerBulkDelete;
-
-export const listDashboards: ListDashboards = () =>
-  DashboardsService.dashboardsControllerList();
-export const createDashboard: CreateDashboard = (dto) =>
-  DashboardsService.dashboardsControllerCreate(dto);
-export const readDashboard: ReadDashboard = (id) =>
-  DashboardsService.dashboardsControllerRead(id);
-export const updateDashboard: UpdateDashboard = (id, dto) =>
-  DashboardsService.dashboardsControllerUpdate(id, dto);
-export const deleteDashboard: DeleteDashboard = (id) =>
-  DashboardsService.dashboardsControllerDelete(id);
-export const bulkDeleteDashboards: BulkDeleteDashboards = (ids) =>
-  DashboardsService.dashboardsControllerBulkDelete(ids);
+export const listDashboards: ListDashboards = isEdgeMode()
+  ? () => listDashboardsLocal()
+  : () => DashboardsService.dashboardsControllerList();
+export const createDashboard: CreateDashboard = isEdgeMode()
+  ? (dto) => createDashboardLocal(dto)
+  : (dto) => DashboardsService.dashboardsControllerCreate(dto);
+export const readDashboard: ReadDashboard = isEdgeMode()
+  ? (id) => readDashboardLocal(id)
+  : (id) => DashboardsService.dashboardsControllerRead(id);
+export const updateDashboard: UpdateDashboard = isEdgeMode()
+  ? (id, dto) => updateDashboardLocal(id, dto)
+  : (id, dto) => DashboardsService.dashboardsControllerUpdate(id, dto);
+export const deleteDashboard: DeleteDashboard = isEdgeMode()
+  ? (id) => deleteDashboardLocal(id)
+  : (id) => DashboardsService.dashboardsControllerDelete(id);
+export const bulkDeleteDashboards: BulkDeleteDashboards = isEdgeMode()
+  ? (dto) => bulkDeleteDashboardsLocal(dto)
+  : (dto) => DashboardsService.dashboardsControllerBulkDelete(dto);
```

Comment:
I think you'll want an additional set of integration tests to ensure `core` is used in the cloud and `localStorage` is used at the edge. Would it make sense to add these now or as a follow-up?

Reply:
Sounds good!